### PR TITLE
FIX: [rar] deadlock with cbr + net vfs (credit notspiff)

### PR DESCRIPTION
**Pull request by koying**
## Description

from notspiff:

specifically we are called off-thread with the main thread waiting on us while holding the gui lock. so it will deadlock on obtaining dialog. the bt was funkily confusing though, as it will actually show a "downstream" deadlock on the rar manager, as there's two concurrent calls to it (to preload/load images). the deadlock that was triggered in the debugger was the second call to the rar manager spinning on the rar manager lock, which is held by the first call to rar manager, which again is spinning on the gui lock.

oh, and using a "exotic" vfs was only part of the problem to the extent that it's slower so the timeout on the progress dialog kicked in.
## Types of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by koying** [comment]
jenkins build this please

File: xbmc/filesystem/RarManager.cpp [hunk 1]
```diff
@@ -88,7 +89,7 @@ class progress_info
   bool progress(int progress, const char *text)
   {
     bool cont(true);
-    if (shown || showTime.IsTimePast())
+    if (shown || showTime.IsTimePast() && g_application.IsCurrentThread())
```
Comment (popcornmix) [inline xbmc/filesystem/RarManager.cpp]: This adds a warnng:
```
RarManager.cpp: In member function 'bool progress_info::progress(int, const char*)':
RarManager.cpp:92:40: warning: suggest parentheses around '&&' within '||' [-Wparentheses]
```

**Comment by koying** [comment]
@popcornmix Bad c/p ;)

**Comment by notspiff** [comment]
.. and i see my request wasn't honored and yet another mail got sent.. :p (i assume you haven't read your mail @koying)

**Comment by koying** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Pull request merged**